tests/s3/test_copy_object.py

```diff
@@ -2257,3 +2261,809 @@ def test_22299(self):
+                                                   self.bucket_name2, self.object_name2)
+        assert_utils.assert_true(status, response)
+        LOGGER.info("Step 3: Copy object to different bucket with condition "                             
+                    "x-amz-copy-source-if-match (False)")
+        try:
+            status, response = self.s3_obj.copy_object(self.bucket_name1, self.object_name1,
+                                                       self.bucket_name2, self.object_name2 + "_im",
+                                                       CopySourceIfMatch=response[1]["ETag"] + "_")
+            assert_utils.assert_false(status, response)
+        except CTException as error:
+            assert_utils.assert_in(errmsg.RGW_ERR_CPY_IF_MATCH_FALSE, error.message, error)
+        LOGGER.info("Step 4: Copy object to different bucket with condition"                              
+                    "x-amz-copy-source-if-none-match (False)")
+        try:
+            status, response = self.s3_obj.copy_object(self.bucket_name1, self.object_name1,
+                                                       self.bucket_name2,
+                                                       self.object_name2 + "_inm",
+                                                        CopySourceIfNoneMatch=response[1]["ETag"])
+            assert_utils.assert_false(status, response)
+        except CTException as error:
+            assert_utils.assert_in(errmsg.RGW_ERR_CPY_IF_NONE_MATCH_FALSE, error.message, error)
+        LOGGER.info("Step 5: Copy object to different bucket with condition "                             
+                    "x-amz-copy-source-if-modified-since (False)")
+        date1 = response[1]["LastModified"]
+        date1 = datetime.strptime(date1, '%b %d %Y')
+        pre_date = date1 - timedelta(days=1)
+        post_date = date1 + timedelta(days=1)
+        try:
+            status, response = self.s3_obj.copy_object(self.bucket_name1, self.object_name1,
+                                                       self.bucket_name2,
+                                                       self.object_name2 + "_ims",
+                                                        CopySourceIfModifiedSince=post_date)
+            assert_utils.assert_false(status, response)
+        except CTException as error:
+            assert_utils.assert_in(errmsg.RGW_ERR_CPY_IF_MODIFIED_SINCE_FALSE, error.message, error)
+        LOGGER.info("Step 6: Copy object to different bucket with condition "                             
+                    "x-amz-copy-source-if-unmodified-since (False)")
+        try:
+            status, response = self.s3_obj.copy_object(
+                self.bucket_name1, self.object_name1, self.bucket_name2,
+                self.object_name2 + "_iums", CopySourceIfUnmodifiedSince=pre_date)
+            assert_utils.assert_false(status, response)
+        except CTException as error:
```

Comment:
The try except can go in helper function with errmsg as one of the parameter

Reply:
I can do that during TE phase as part of RVF